### PR TITLE
solve perf issue and add sent_date where pertinent

### PR DESCRIPTION
**Pull request by monodo**

### CODE REVIEW

File: geocity/apps/submissions/models.py [hunk 1]
```diff
@@ -211,6 +213,8 @@ class Submission(models.Model):
     )
     created_at = models.DateTimeField(_("date de création"), default=timezone.now)
     validated_at = models.DateTimeField(_("date de validation"), null=True)
+    # TODO: rename on next API break => (last_sent_at)
```
Comment (rbovard) [inline geocity/apps/submissions/models.py:216]: Are you sure it's neccessary?

File: geocity/apps/submissions/migrations/0018_change_sent_date_into_field.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Generated by Django 4.2.1 on 2023-06-05 11:55
+
+from django.db import migrations, models
+
+
+def set_sent_date_from_simple_history(apps, schema_editor):
```
Comment (rbovard) [inline geocity/apps/submissions/migrations/0018_change_sent_date_into_field.py:6]: There is an issue with the migration:

```
Running migrations:
  Applying submissions.0018_change_sent_date_into_field...Traceback (most recent call last):
  File "/code/manage.py", line 15, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.10/dist-packages/django/core/management/__init__.py", line 442, in execute_from_command_line  
    utility.execute()
  File "/usr/local/lib/python3.10/dist-packages/django/core/management/__init__.py", line 436, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.10/dist-packages/django/core/management/base.py", line 412, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.10/dist-packages/django/core/management/base.py", line 458, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.10/dist-packages/django/core/management/base.py", line 106, in wrapper
    res = handle_func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/django/core/management/commands/migrate.py", line 356, in handle
    post_migrate_state = executor.migrate(
  File "/usr/local/lib/python3.10/dist-packages/django/db/migrations/executor.py", line 135, in migrate
    state = self._migrate_all_forwards(
  File "/usr/local/lib/python3.10/dist-packages/django/db/migrations/executor.py", line 167, in _migrate_all_forwards        
    state = self.apply_migration(
  File "/usr/local/lib/python3.10/dist-packages/django/db/migrations/executor.py", line 252, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/usr/local/lib/python3.10/dist-packages/django/db/migrations/migration.py", line 132, in apply
    operation.database_forwards(
  File "/usr/local/lib/python3.10/dist-packages/django/db/migrations/operations/special.py", line 193, in database_forwards  
    self.code(from_state.apps, schema_editor)
  File "/code/geocity/apps/submissions/migrations/0018_change_sent_date_into_field.py", line 11, in set_sent_date_from_simple_history
    row.history.filter(status=Submission.STATUS_SUBMITTED_FOR_VALIDATION)
AttributeError: 'Submission' object has no attribute 'history'
```

**Review by rbovard — CHANGES_REQUESTED** [review]

File: geocity/apps/submissions/migrations/0018_change_sent_date_into_field.py (hunk 2 above)
Comment (monodo) [reply geocity/apps/submissions/migrations/0018_change_sent_date_into_field.py]: Could you please pull & try again ?

**Review by rbovard — DISMISSED** [review]
Hum, the migration no longer crashes, but the result is not as expected. On my preprod, only the submissions with `STATUS_SUBMITTED_FOR_VALIDATION` have a sent date, not the other ones. All the submission exept `STATUS_DRAFT` should have a sent_date.

And if an user only has drafts, they are dispayed in the wrong order. Please add extra order field on the creation date to also sort drafts.

**Comment by monodo** [comment]
> `STATUS_SUBMITTED_FOR_VALIDATION`



> Hum, the migration no longer crashes, but the result is not as expected. On my preprod, only the submissions with `STATUS_SUBMITTED_FOR_VALIDATION` have a sent date, not the other ones. All the submission exept `STATUS_DRAFT` should have a sent_date.
> 
> And if an user only has drafts, they are dispayed in the wrong order. Please add extra order field on the creation date to also sort drafts.

Fixed

File: geocity/apps/submissions/search.py [hunk 3]
```diff
@@ -76,7 +76,7 @@ def match_type_label(match_type):
     return {
         MatchType.AUTHOR: _("Auteur‧e"),
         MatchType.FIELD: _("Propriété"),
-        MatchType.CREATED_AT: _("Date de création"),
+        MatchType.SENT_DATE: _("Date de création"),
```
Comment (AlexandreJunod) [inline geocity/apps/submissions/search.py:79]: Text is no longer `_("Date de création"),`

File: geocity/apps/submissions/templates/submissions/submission_delete.html [hunk 4]
```diff
@@ -11,7 +11,7 @@ <h4>{% translate "Détails de la demande:" %} </h4>
     <ul>
       <li>{% translate "Requérant:"%} {{ submission.author.first_name }}, {{ submission.author.last_name}}
       </li>
-      <li>{% translate "Demande du:"%} {{ submission.created_at }}</li>
+      <li>{% translate "Demande du:"%} {{ submission.sent_date }}</li>
```
Comment (AlexandreJunod) [inline geocity/apps/submissions/templates/submissions/submission_delete.html:14]: Change `Demande du` ?

**Review by AlexandreJunod — CHANGES_REQUESTED** [review]

**Pull request merged**